ide-support/revliburl.livecodescript

```diff
@@ -5290,6 +5309,345 @@ private function _proxyfromHTTPProxy pURL, pProxy
+         
+         ## Create array of cookie entries
+         delete char 1 to length("Set-Cookie:") of tCookieLine 
+         put word 1 to -1 of tCookieLine into tCookieLine ## strip white space
+         split tCookieLine by ";" and "="
+         repeat for each key tKey in tCookieLine
+            put word 1 to -1 of tCookieLine[tKey] into tCookieA[word 1 to -1 of tKey]
+         end repeat
+         
+         ## Session cookies that specify a domain are rejected
+         if tCookieA["expires"] is empty and tCookieA["domain"] is not empty then
+            ------------
+            put "Cookie doesn't expire and has a domain:" into tMsg
+            repeat for each key tKey in tCookieA
+               put cr & tKey & ":" && tCookieA[tKey] after tMsg
+            end repeat
+            ulLogIt tMsg & cr
+            -------------
+            next repeat
+         end if
+         
+         ## Validate domain
+         if tCookieA["domain"] is empty then
+            ## No domain, using current
+            put tHost into tCookieA["domain"]
+            
+            ## must have at least 2 "."
+            ## tack on "." to beginning if need be (i.e. domain is "google.com" rather than "www.google.com")
+            put offset(".", tCookieA["domain"]) into tCharNo
+            if tCharNo > 0 then put offset(".", tCookieA["domain"], tCharNo) into tCharNo
+            if tCharNo is 0 then
+               put "." before tCookieA["domain"]
+            end if
+         end if
+         
+         ## must have at least 2 "."
+         put offset(".", tCookieA["domain"]) into tCharNo
+         if tCharNo > 0 then
+            put offset(".", tCookieA["domain"], tCharNo) into tCharNo
+         end if
+         
+         if tCharNo is 0 then
+            put "Cookie domain does not contain at least 2 '.' :" into tMsg
+            repeat for each key tKey in tCookieA
+               put cr & tKey & ":" && tCookieA[tKey] after tMsg
+            end repeat
+            ulLogIt tMsg & cr
+            next repeat
+         end if
+         
+         ## Check that tCookieA["domain"] is part of tHost
+         put tCookieA["domain"] into tCookieDomain
+         if char 1 of tCookieDomain is "." then delete char 1 of tCookieDomain
+         if char -length(tCookieDomain) to -1 of tHost is not tCookieDomain then next repeat
+         
+         ## Normalize
+         if tCookieA["path"] is empty then put tDefaultPath into tCookieA["path"]
+         if tCookieA["expires"] is not empty then
+            ## Store expires in seconds
+            put _ConvertGMTInternetTimestampToInternetDate(tCookieA["expires"]) into tCookieA["expires"]
+            convert tCookieA["expires"] from internet date to seconds
+         end if
+         put "secure" is among the keys of tCookieA into tCookieA["secure"]
+         
+         ## Store each unique name=value pair on a separate line.
+         ## This allows us to easily update an existing cookie later on
```

Comment:
Have you considered making `laCookies` a multi-dimensional array?  For example, `laCookies[<domain>]["expires"]`, `laCookies[<domain>]["path"]`, etc.  Do you think it would make the code simpler than having to encode the information in a string and then parse it again each time you want to check it?  I can see an argument either way, because at the moment you can put the contents of `laCookies[<domain>]` directly into a `Cookie:` header when making a request.